pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -761,10 +761,19 @@ func (r *ReconcileBareMetalHost) actionManageHostPower(prov provisioner.Provisio
 
 // Adopt the host if necessary and manage the current power status.
 func (r *ReconcileBareMetalHost) actionManageExternallyProvisioned(prov provisioner.Provisioner, info *reconcileInfo) (result reconcile.Result, err error) {
+
 	provResult, err := prov.Adopt()
 	if err != nil {
 		return
 	}
+	if provResult.ErrorMessage != "" {
+		info.host.Status.Provisioning.State = metal3v1alpha1.StateRegistrationError
+		if info.host.SetErrorMessage(provResult.ErrorMessage) {
+			info.publishEvent("RegistrationError", provResult.ErrorMessage)
+			result.Requeue = true
+		}
+		return result, nil
```

Comment:
If we've failed to adopt the host, I don't think we want to worry about the power anyway, do we? Doesn't a failure here imply that we have bad credentials?

Reply:
It doesn't imply that AFAIK (currently we're failing because we're not supplying the Image ID, for example, but even once we've fixed that I assume there are other ways it could fail).
We manage the power state on servers that haven't been adopted, so it seems like we should at least attempt to do that when we've failed to adopt.

Reply:
Looking at the ironic code, the only reasons I can see for a failure are bad credentials or a bug in ironic itself. https://opendev.org/openstack/ironic/src/branch/master/ironic/conductor/manager.py#L2037

Reply:
I concur with @dhellmann. Essentially adoption will require the image fields in the event it is asked to rebuild a host... and otherwise only bmc connectivity is checked. I mean, you could have a transitory RPC failure or something deep under the hood, but trying to manage power state or change it is ill advised and may still fail.

Also, existing power state will not be known until adoption is complete, so it seems kind of odd to try and perform a power action upon failure.

Reply:
What I'm hearing is that we shouldn't try to manage the power state of an externally-provisioned host that hasn't been adopted at all?

Reply:
I think that's right, and I think that's what will happen now.

Reply:
@zaneb That is correct.